File: Selenium/Project/Activity8.java
package project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.time.Duration;

public class Activity8 {

    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void beforeClass() {
        // Set up the Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Initialize Firefox driver object
        driver = new FirefoxDriver();
        // Initialize wait object
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        //Open browser
        driver.get("https://alchemy.hguy.co/jobs/wp-admin");
    }

    @Test
    public void testLogin() {
        WebElement username = driver.findElement(By.id("user_login"));
        WebElement password = driver.findElement(By.id("user_pass"));
        WebElement submit = driver.findElement(By.id("wp-submit"));
        username.sendKeys("root");
        password.sendKeys( "pa$$w0rd");
        submit.click();

        //Assertion
        Assert.assertEquals(driver.getTitle(),"Dashboard ‹ Alchemy Jobs — WordPress");
    }
    @AfterClass
    public void afterMethod() {
        //Close the browser
        driver.quit();
    }

}

File: Selenium/Project/Activity8.py
import time
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By

service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("http://alchemy.hguy.co/orangehrm")
    # Print the title of the page
    print("Page title is: ", driver.title)

    # Find the username field
    Username = driver.find_element(By.ID, "txtUsername")
    # Find the password field
    Password = driver.find_element(By.ID, "txtPassword")

    # Enter the given credentials
    # Enter username
    Username.send_keys("orange")
    # Enter password
    Password.send_keys("orangepassword123")

    #click the login button
    driver.find_element(By.ID, "btnLogin").click()
    driver.find_element(By.LINK_TEXT,"Dashboard").click()
    driver.find_element(By.LINK_TEXT,"Apply Leave").click()
    driver.find_element(By.ID,"applyleave_txtLeaveType").click()
    driver.find_element(By.ID,"applyleave_txtLeaveType").send_keys("Dayoff")
    driver.find_element(By.ID,"applyleave_txtFromDate").click()
    driver.find_element(By.CLASS_NAME,"ui-datepicker-trigger").click()
    driver.find_element(By.ID,"applyleave_txtFromDate").send_keys("2023-12-01")
    driver.find_element(By.ID,"applyBtn").click()
    time.sleep(100)
    driver.find_element(By.ID,"menu_leave_viewMyLeaveList").click()
    driver.close()
